R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -848,24 +848,31 @@ test_that("collect() and take() on a DataFrame return the same number of rows an
 })
 
 test_that("collect() support Unicode characters", {
-  lines <- c("{\"name\":\"안녕하세요\"}",
-             "{\"name\":\"您好\", \"age\":30}",
-             "{\"name\":\"こんにちは\", \"age\":19}",
-             "{\"name\":\"Xin chào\"}")
+  jsonPath <- file.path(
+    Sys.getenv("SPARK_HOME"),
+    "R", "pkg", "tests", "fulltests", "data",
+    "test_utils_utf.json"
+  )
+
+  lines <- readLines(jsonPath, encoding = "UTF-8")
 
-  jsonPath <- tempfile(pattern = "sparkr-test", fileext = ".tmp")
-  writeLines(lines, jsonPath)
+  expected <- regmatches(lines, gregexpr('(?<="name": ").*?(?=")', lines, perl = TRUE))
 
   df <- read.df(jsonPath, "json")
   rdf <- collect(df)
   expect_true(is.data.frame(rdf))
-  expect_equal(rdf$name[1], markUtf8("안녕하세요"))
```

Comment:
Ur, this is a required change for `testthat`?

Reply:
This fails on Windows due to encoding issues, irrespective of testthat version. Similar to recursive call test case, failure is easy to reproduce (Windows, English locale,  CP1252 encoding) outside tests, but somehow is ignored in testthat 1.x.

@dongjoon-hyun The alternative to this is to write bytes explicitly, i.e.

```
test_that("collect() support Unicode characters", {

  lines <- markUtf8(c(
    '{"name": "안녕하세요"}',
    '{"name": "您好", "age": 30}',
    '{"name": "こんにちは", "age": 19}',
    '{"name": "Xin ch\xc3\xa0o"}'
  ))

  jsonPath <- tempfile(pattern = "sparkr-test", fileext = ".tmp")
  writeLines(lines, jsonPath, useBytes = TRUE)

  expected <- regmatches(lines, regexec('(?<="name": ").*?(?=")', lines, perl = TRUE))

  df <- read.df(jsonPath, "json")
  rdf <- collect(df)
  expect_true(is.data.frame(rdf))

  rdf$name <- markUtf8(rdf$name)
  expect_equal(rdf$name[1], expected[[1]])
  expect_equal(rdf$name[2], expected[[2]])
  expect_equal(rdf$name[3], expected[[3]])
  expect_equal(rdf$name[4], expected[[4]])

  df1 <- createDataFrame(rdf)
  expect_equal(
    collect(
      where(df1, df1$name == expected[[2]])
    )$name,
    expected[[2]]
  )
})
```

or skip test case on Windows, but as-is, this will and should fail on AppVeyor.

Reply:
Please make another PR for that. One PR should have one theme.
>  irrespective of testthat version

Reply:
So you're saying that we should accept invalid test failures here? Or fix this beforehand on 1.x, even if problem is not visible with 1.x?

Reply:
Yes, right. That would be great.

Reply:
For Windows, @HyukjinKwon is the best person. So, he can override my comment.

Reply:
Do you have any preference towards the method used?

In general I have no access to Windows setup on which I can test this across testthat (things vary, as 2.0 and latest 2.x versions come with different dependencies), R (we have 3.1 on Jenkins, latest on Windows builds) and encoding. So this is the best I came up with, in the limited time I had to tinker with the whole thing. But I am opened to other proposals.

Reply:
Ur, if you are unable to verify, please don't make a PR.
> I have no access to Windows setup

Reply:
For the reference I've opened [SPARK-30645](https://issues.apache.org/jira/browse/SPARK-30645) to track this one.